scipy/optimize/_qap.py

```diff
@@ -0,0 +1,389 @@
+           matching," PLOS one, vol. 10, no. 4, p. e0121002, 2015.
+
+    .. [2] D. Fishkind, S. Adali, H. Patsolic, L. Meng, D. Singh, V. Lyzinski,
+           C. Priebe, Seeded graph matching, Pattern Recognit. 87 (2019)
+           203–215
+
+    Examples
+    --------
+
+    >>> cost = [[0,22,53,53],[22,0,40,62],[53,40,0,55],[53,62,55,0]]
+    >>> dist = [[0,3,0,2],[3,0,0,1],[0,0,0,4],[2,1,4,0]]
+    >>> from scipy.optimize import quadratic_assignment
+    >>> quadratic_assignment(cost,dist)
+       col: array([2, 3, 0, 1])
+       nit: 30
+       row: array([0, 1, 2, 3])
+     score: 790
+
+    """
+
+    cost_matrix = np.asarray(cost_matrix)
+    dist_matrix = np.asarray(dist_matrix)
+    seed_cost = np.asarray(seed_cost)
+    seed_dist = np.asarray(seed_dist)
+
+    if cost_matrix.shape[0] != dist_matrix.shape[0]:
+        msg = "Adjacency matrices must be of equal size"
+        raise ValueError(msg)
+    elif (
+        cost_matrix.shape[0] != cost_matrix.shape[1]
+        or dist_matrix.shape[0] != dist_matrix.shape[1]
+    ):
+        msg = "Adjacency matrix entries must be square"
+        raise ValueError(msg)
+    elif not (cost_matrix >= 0).all() or not (dist_matrix >= 0).all():
+        msg = "Adjacency matrix entries must be greater than or equal to zero"
+        raise ValueError(msg)
+    elif seed_cost.shape[0] != seed_dist.shape[0]:
+        msg = "Seed arrays must be of equal size"
+        raise ValueError(msg)
+    elif seed_cost.shape[0] > cost_matrix.shape[0]:
+        msg = "There cannot be more seeds than there are nodes"
+        raise ValueError(msg)
+    elif not (seed_cost >= 0).all() or not (seed_dist >= 0).all():
+        msg = "Seed array entries must be greater than or equal to zero"
+        raise ValueError(msg)
+    elif (
+        not (seed_cost <= (cost_matrix.shape[0] - 1)).all()
+        or not (seed_dist <= (cost_matrix.shape[0] - 1)).all()
+    ):
+        msg = "Seed array entries must be less than or equal to n-1"
+        raise ValueError(msg)
+    elif type(n_init) is not int and n_init <= 0:
+        msg = '"n_init" must be a positive integer'
+        raise TypeError(msg)
+    elif not init_method == "barycenter" and not init_method == "rand":
+        msg = 'Invalid "init_method" parameter string'
+        raise ValueError(msg)
+    elif max_iter <= 0 and type(max_iter) is not int:
+        msg = '"max_iter" must be a positive integer'
+        raise TypeError(msg)
+    elif type(shuffle_input) is not bool:
+        msg = '"shuffle_input" must be a boolean'
+        raise TypeError(msg)
+    elif eps <= 0 and type(eps) is not float:
+        msg = '"eps" must be a positive float'
+        raise TypeError(msg)
+    elif type(maximize) is not bool:
+        msg = '"maximize" must be a boolean'
+        raise TypeError(msg)
+
+    n = cost_matrix.shape[0]  # number of vertices in graphs
+    n_seeds = seed_cost.shape[0]  # number of seeds
+    n_unseed = n - n_seeds
+
+    score = math.inf
```

Comment:
Very nitpicky, sorry: consider moving this down under `obj_func_scalar` to maintain the order used under `if maximize`/